### PR TITLE
bugs fix for share case

### PR DESCRIPTION
**Pull request by rayliangatos**
Resolves # . (This is applicable only if this pull request relates to a GitHub issue, delete the line otherwise)

Notes:
*EUI-2409:diable add user button if no selected case

### CODE REVIEW

File: projects/exui-common-lib/src/lib/components/selected-case/selected-case.component.ts [hunk 1]
```diff
@@ -40,7 +42,13 @@ export class SelectedCaseComponent implements OnInit {
     return user.idamId;
   }
 
+  public getFeatureToggleFlag(): Observable<boolean> {
```
Comment (udaydenduluri33) [inline projects/exui-common-lib/src/lib/components/selected-case/selected-case.component.ts:45]: @rayliangatos  we cannot have feature toggle code in library. This code has to be moved to application. Thanks

File: projects/exui-common-lib/src/lib/components/selected-case/selected-case.component.ts [hunk 2]
```diff
@@ -56,10 +64,9 @@ export class SelectedCaseComponent implements OnInit {
         }
       }
     });
-    return canRemove;
+    return removeUserFromCaseToggleOn && canRemove;
```
Comment (ronaldmansveld) [inline projects/exui-common-lib/src/lib/components/selected-case/selected-case.component.ts:67]: This leads to race conditions, as you're relying on the fact that the subscription finished before you reach this line. This method should return an Observable<boolean> itself, and use the two observables you subscribe to now to combine them to create the final value, and return that in an Observable.

File: projects/exui-common-lib/src/lib/components/selected-case/selected-case.component.spec.ts [hunk 3]
```diff
@@ -2,19 +2,26 @@ import { async, ComponentFixture, TestBed } from '@angular/core/testing';
...
     TestBed.configureTestingModule({
-      declarations: [ SelectedCaseComponent ]
+      declarations: [ SelectedCaseComponent ],
+      providers: [{
+        provide: FeatureToggleService,
+        useValue: mockFeatureToggleService
+      }]
     })
     .compileComponents();
+    mockFeatureToggleService.getValue.and.returnValue(of(true));
```
Comment (Mo-Lala84) [inline projects/exui-common-lib/src/lib/components/selected-case/selected-case.component.spec.ts:24]: we need to abstract this flag out as discussed in our design approach in following the twelve factor principle for web development..we cant be having to rebuild the application each time we need to do a change...please move out from the app into a config which we can drive through our custom config which are examples in code already.

File: projects/exui-common-lib/src/lib/components/selected-case/selected-case.component.ts (hunk 2 above)
Comment (rayliangatos) [reply projects/exui-common-lib/src/lib/components/selected-case/selected-case.component.ts]: moved the flag to webapp as per other comments.

File: projects/exui-common-lib/src/lib/components/selected-case/selected-case.component.ts (hunk 1 above)
Comment (rayliangatos) [reply projects/exui-common-lib/src/lib/components/selected-case/selected-case.component.ts]: moved the flag to webapp now

File: projects/exui-common-lib/src/lib/components/selected-case/selected-case.component.spec.ts (hunk 3 above)
Comment (rayliangatos) [reply projects/exui-common-lib/src/lib/components/selected-case/selected-case.component.spec.ts]: moved the flag to webapp as per other comments.

File: projects/exui-common-lib/src/lib/components/selected-case/selected-case.component.ts [hunk 4]
```diff
@@ -56,10 +57,9 @@ export class SelectedCaseComponent implements OnInit {
         }
       }
     });
-    return canRemove;
+    return this.removeUserFromCaseToggleOn && canRemove;
```
Comment (ronaldmansveld) [inline projects/exui-common-lib/src/lib/components/selected-case/selected-case.component.ts:60]: Not sure where the above code comes from, but this still has the racecondition in it, so this needs a refactor.

Comment (ronaldmansveld) [reply projects/exui-common-lib/src/lib/components/selected-case/selected-case.component.ts:60]: Same goes for the methods below. You can't subscribe to an observable and run that code immediately to return something synchronously.

**Pull request merged**